### PR TITLE
fixes timezone issue in date of birth in patient registration

### PR DESCRIPTION
**Pull request by rithviknishad**
- [ENG-288]

### Bug

Users cannot register a patient on today’s date if attempted to do between 12 AM and 5:30 AM IST (or whichever is the deployment’s timezone offset) as this shows “Date cannot be in the future” front-end error.

[ENG-288]: https://openhealthcarenetwork.atlassian.net/browse/ENG-288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved date-of-birth validation to correctly handle timezone edge cases so same-day registrations succeed and future dates are rejected.
  * Fixed device manufacture-date validation to reliably reject future dates.

* **Tests**
  * Added end-to-end tests covering DOB timezone validation, including same-day registrations and future-date rejection across timezones.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ohcnetwork/care_fe/pull/16347?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/components/Patient/PatientRegistration.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ import validators from "@/Utils/validators";
 import careConfig from "@careConfig";
 import { zodResolver } from "@hookform/resolvers/zod";
 import { useMutation, useQuery } from "@tanstack/react-query";
-import { format, isBefore, isFuture, subYears } from "date-fns";
+import { format, isBefore, subYears } from "date-fns";
```
Comment (bodhish) [inline src/components/Patient/PatientRegistration.tsx:70]: Why don't we cleanup all uses of `isFuture` 🤔

**Review by bodhish — CHANGES_REQUESTED** [review]

**Comment by parvathyns-creator** [comment]
@nihal467 this looks fine

**Review by nihal467 — APPROVED** [review]

File: tests/facility/patient/patientRegistration.spec.ts [hunk 2]
```diff
@@ -264,3 +264,62 @@ test.describe("Patient Registration", () => {
...
+
+    await page.goto("/");
+    await page
+      .getByRole("link", { name: /facility with patients/i })
+      .first()
+      .click();
+    await page.getByRole("button", { name: "Toggle Sidebar" }).click();
+    await page.getByRole("button", { name: "Patients", exact: true }).click();
+    await page.getByRole("link", { name: /search patients/i }).click();
+  });
```
Comment (yash-learner) [inline tests/facility/patient/patientRegistration.spec.ts]: Why not do a direct navigation here ?

We already have `facility ID` stored via facilityId.ts in test support 

```ts
  await page.goto(
    `/facility/${facilityId}/patient/create`,
  );
````

**Review by yash-learner — CHANGES_REQUESTED** [review]

**Review by yash-learner — APPROVED** [review]
LGTM
